carbon-design-system/carbon-for-ibm-dotcom
# feat(card): add logo variant

### Related Ticket(s)

DDS Consulting Issue: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7139
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-2312

### Description
- This PR adds a new variant within the `Card` component called `Card logo`. 

### Changelog

**New Components/Props**

- `<dds-image-logo>` - this component extends from `<dds-image>` and is needed because of styling requirements.

**Removed**

- N/A

### Testing Instructions

- Visit the `Card` UI component with `Logo` variant - `?path=/story/components-card--logo`
- Ensure that the preview matches with the design specs 
  - [Design and functional spec Box folder](https://ibm.ent.box.com/folder/95514001950?s=hgvu48r1fv2ga2h0yn3qshrtxjc71vch)
